external/js/cothority/lib/identity.js

```diff
@@ -0,0 +1,227 @@
+"use strict";
+
+const topl = require('topl');
+const UUID = require('pure-uuid');
+const protobuf = require('protobufjs')
+
+const misc = require('./misc');
+
+
+/**
+ * ServerIdentity represents a cothority server. It has an associated public key
+ * and a TCP address. The WebSocket address is derived from the TCP address.
+ * */
+class ServerIdentity {
+    /*
+     * Returns a new ServerIdentity from the public key and address.
+     * @param {Uint8Array} public key in bytes format
+     * @param {string} address tcp address of the cothority node
+     * @param {string} description of the conode. Can be null.
+     * @return a ServerIdentity
+     * */
+    constructor(publicKey, address,description) {
+        if (publicKey.constructor !== Uint8Array)
+            throw TypeError;
+        this.pub = publicKey
+        this.addr = address;
+        this._description = description;
+        // id of the identity
+        const url = 'https://dedis.epfl.ch/id/' + misc.uint8ArrayToHex(this.pub);
+        this._id = new UUID(5, 'ns:URL', url).export();
+        // tcp + websocket address
+        let parts = address.replace("tcp://", "").split(":");
```

Comment:
address will shortly start with tls:// and this will break.

Reply:
I now split via the `://` separation "word".